### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by Amazinggracee**
### **Project requirements**
**Design**

- [ ] You should follow these design guidelines, including:

- Colors.
- Typography: font face, size and weight.
- Layout: composition and space between elements.

Original design idea by Gregoire Vella on Behance.

- The Creative Commons license of the design requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**

- [ ] Splash screen
- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- [ ] Sign up and log in pages
- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ] Home page (categories page)
- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [ ] Transactions page
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new transaction" page

- [ ] Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: app/views/balances/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            <p class="balance-date"><%= balance.created_at.strftime('%d-%b-%Y') %></p>
+          </div>
+          <p class="balance-amount">$<%= balance.items.map { |item| item.amount }.sum %></p>
+        </li>
+      <% end %>
+    <% end %>
+  </ul>
+</div>
+
+<div class="add-button">
```
Comment (t-yanick) [inline app/views/balances/index.html.erb:37]: - [ ] Great job working on this capstone. As per project requirements, you are supposed to follow the `layout` in the given design. At the moment, your `add button` does not follow the given layout. See screenshots below:
- Your design
Screen Shot 2023-09-14 at 7 44 23 PM
- Given design
Screen Shot 2023-09-14 at 7 46 56 PM

The add button should be at the button and the length should be the width of the page

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Amazinggracee , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Excellent video presentation within the time limit.
:heavy_check_mark: The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
:heavy_check_mark: The video contains a demo of the project, highlighting all the project features. kudos for the piece of code your are proud of making.
:heavy_check_mark: Deployment done on Render
:heavy_check_mark: Sign up and login pages present and a user can register and log in successfully
:heavy_check_mark: All pages added as per project requirements
:heavy_check_mark: Postgres used for database and devise for authentication
:heavy_check_mark: App has unit and integration tests for all the most important components of the RoR application.
:heavy_check_mark: App works as expected

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/balances/index.html.erb (hunk 1 above)
Comment (Amazinggracee) [reply app/views/balances/index.html.erb:37]: @t-yanick , how are you doing today? Thank you for pointing this out. I have changed the design.

**Review by Olamarx — APPROVED** [review]
Hi @Amazinggracee,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Required changes fixed ✔️
- Commit messages meaningful ✔️
- Gitflow ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**